### PR TITLE
Add config files for Kuro65 keyboard

### PR DESCRIPTION
**Pull request by 0x544D**
## Description

This PR will bring in configuration files and keymaps for the open-source keyboard Kuro65.
The Kuro65 keyboard is a 65% keyboard with an ISO layout, which fits the TOFU65 case (and similar) by KBDFans.
Features include:

- Hot-swappable ISO layout
- Per key RGB LED (south facing)
- VIA compatible
- USB-C connector
- Reset button on the back of the PCB
- Compatible with TOFU65 and similar cases 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by 0x544D** [comment]
I will add a link to the PR for the VIA repository, as soon as I created it.

File: keyboards/kuro/kuro65/config.h [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
```
Comment (waffle87) [inline keyboards/kuro/kuro65/config.h:19]:
```suggestion
```

File: keyboards/kuro/kuro65/config.h [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+/* RGB Matrix defines*/
+#ifdef RGB_MATRIX_ENABLE
+// The pin connected to the data pin of the LEDs
+#    define RGB_DI_PIN E6
+// The number of LEDs connected
+#    define RGB_MATRIX_LED_COUNT 69
+
+#    define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
```
Comment (waffle87) [inline keyboards/kuro/kuro65/config.h:28]:
```suggestion
```

File: keyboards/kuro/kuro65/config.h [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#    define ENABLE_RGB_MATRIX_SPLASH
+#    define ENABLE_RGB_MATRIX_MULTISPLASH
+#    define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#    define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+
+#    define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_ALL
```
Comment (waffle87) [inline keyboards/kuro/kuro65/config.h:83]:
```suggestion
#    define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_CYCLE_ALL
```

File: keyboards/kuro/kuro65/keymaps/0x544D/keymap.c [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  */
+
+#include QMK_KEYBOARD_H
+
+#define BL_WIN 0 // base layer Windows
+#define BL_MAC 1 // base layer Mac OS
+#define FN1    2 // function layer 1
+#define FN2    3 // function layer 2
+
```
Comment (waffle87) [inline keyboards/kuro/kuro65/keymaps/0x544D/keymap.c:23]:
```suggestion
enum layers {
    BL_WIN,
    BL_MAC,
    FN1,
    FN2
};
```

File: keyboards/kuro/kuro65/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  * You should have received a copy of the GNU General Public License
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  */
+
+#include QMK_KEYBOARD_H
+
+#define BASE   0 // base layer Windows
+#define FN1    1 // function layer 1
+#define FN2    2 // function layer 2
+
```
Comment (waffle87) [inline keyboards/kuro/kuro65/keymaps/default/keymap.c:22]:
```suggestion
enum layers {
    BASE,
    FN1,
    FN2
};
```

File: keyboards/kuro/kuro65/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  * You should have received a copy of the GNU General Public License
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  */
+
+#include QMK_KEYBOARD_H
+
+#define BASE   0 // base layer Windows
+#define FN1    1 // function layer 1
+#define FN2    2 // function layer 2
+
```
Comment (waffle87) [inline keyboards/kuro/kuro65/keymaps/via/keymap.c:22]:
```suggestion
enum layers {
    BASE,
    FN1,
    FN2
};
```

File: keyboards/kuro/kuro65/readme.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+- VIA compatible
+- USB-C connector
+- Reset button on the back of the PCB
+- Compatible with TOFU65 and similar cases 
+
+## Pictures
+
+![kuro65_pcb](https://user-images.githubusercontent.com/28592326/225080112-e31c2d8d-801e-48e6-b4a8-996fe5636419.jpeg)
+
+![kuro65_case](https://user-images.githubusercontent.com/28592326/225080143-74d2be9f-fc1b-4c57-8ad6-8ea516279ff8.jpeg)
```
Comment (waffle87) [inline keyboards/kuro/kuro65/readme.md:21]: These images should be rehosted on Imgur. And generally HTML tags should be avoided in markdown.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/kuro/kuro65/readme.md (hunk 7 above)
Comment (0x544D) [reply keyboards/kuro/kuro65/readme.md]: I hosted all images on imgur and adjusted the README.md accordingly.

Thank you very much for the review!

**Comment by 0x544D** [comment]
Addressed all change requests of reviewer. Thank you very much for the review!

File: keyboards/kuro/kuro65/info.json [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+            { "flags": 4, "matrix": [0, 11], "x":165, "y":  0}, 
+            { "flags": 4, "matrix": [0, 12], "x":180, "y":  0}, 
+            { "flags": 4, "matrix": [0, 14], "x":203, "y":  0}, 
+            { "flags": 4, "matrix": [0, 15], "x":224, "y":  0}
+        ]
+    },
+    "bootmagic": { 
+        "enabled": true,
+        "matrix": [0, 0]
+    }
```
Comment (waffle87) [inline keyboards/kuro/kuro65/info.json:190]: This is the default behaviour
```suggestion
    }
```

Comment (0x544D) [reply keyboards/kuro/kuro65/info.json]: Thanks! I was not sure, if I need this section.

**Comment by 0x544D** [comment]
Removed bootmagic section from info.json as requested, since specific configuration was default behavior.
Thanks!

**Review by waffle87 — APPROVED** [review]

File: keyboards/kuro/kuro65/keymaps/0x544D/keymap.c [hunk 9]
```diff
@@ -0,0 +1,102 @@
+/* Copyright 2023 Tobias Minn (0x544D)
```
[the remark this answers is not in the document]
Comment (0x544D) [reply keyboards/kuro/kuro65/keymaps/0x544D/keymap.c]: Renamed keymap folder to lowercase, as requested. Thanks !

**Comment by 0x544D** [comment]
Applied all requested changes. @lesshonor Thank you very much for the review !

**Review by drashna — APPROVED** [review]

**Pull request merged**